src/interface/pipe.rs

```diff
@@ -1,8 +1,14 @@
+#![allow(dead_code)]
 // Author: Nicholas Renner
-//
-// Pipes for SafePOSIX based on Lock-Free Circular Buffer
 
-#![allow(dead_code)]
+//! In-Memory Pipe Imlpementation for the RustPOSIX interface
+//!
+//! ## Pipe Module
+//!
+//! This module provides a method for in memory iPC between cages and is able to replicate both pipes and Unix domain sockets
```

Comment:
Could use about 2-3x more detail.  Are you using a ring buffer of pages or bytes?  How does this differ from normal pipes?  This is UDS also...  how does that plug in and what things are and are not implemented? , etc.